code/modules/research/rdconsole.dm

```diff
@@ -302,12 +302,12 @@ won't update every console in existence) but it's more of a hassle to do. Also,
 							if(linked_destroy.loaded_item.reliability < 100 && linked_destroy.loaded_item.crit_fail)
 								files.UpdateDesign(linked_destroy.loaded_item.type)
 							if(linked_lathe) //Also sends salvaged materials to a linked protolathe, if any.
-								var/datum/material/metal = linked_lathe.materials["iron"]
-								var/datum/material/glass = linked_lathe.materials["glass"]
+								var/datum/material/metal = linked_lathe.materials.getMaterial("iron")
+								var/datum/material/glass = linked_lathe.materials.getMaterial("glass")
 								metal.stored += min((linked_lathe.max_material_storage - linked_lathe.TotalMaterials()), (linked_destroy.loaded_item.m_amt*linked_destroy.decon_mod))
 								glass.stored += min((linked_lathe.max_material_storage - linked_lathe.TotalMaterials()), (linked_destroy.loaded_item.g_amt*linked_destroy.decon_mod))
-								linked_lathe.materials["iron"]=metal
-								linked_lathe.materials["glass"]=glass

```

Comment:
because jesus fuck Comic made a mess out of this and I'm desperately going through all of this and he will have to do the same when he gets back
if that turns out to be needed later, it's still there